### PR TITLE
Test self.event in workers

### PR DESCRIPTION
**Pull request by annevk**
To compliment #10329. Needed for https://github.com/whatwg/dom/pull/407.

### CODE REVIEW

File: dom/events/event-global.worker.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+importScripts("/resources/testharness.js");
+test(t => {
+  let seen = false;
+  const event = new Event("hi");
+  assert_equals(self.event, undefined);
+  self.addEventListener("hi", t.step_func(e => {
```
Comment (foolip) [inline dom/events/event-global.worker.js]: I think it's harmless, but the `t.step_func` isn't needed for synchronous tests like this one.

File: dom/events/event-global.worker.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    seen = true;
+    assert_equals(self.event, undefined);
+    assert_equals(e, event);
+  }));
+  self.dispatchEvent(event);
+  assert_true(seen);
+}, "There's no self.event (that's why we call it window.event) in workers");
```
Comment (foolip) [inline dom/events/event-global.worker.js]: I like this test title :)

**Review by foolip — APPROVED** [review]
LGTM % optional simplification

File: dom/events/event-global.worker.js (hunk 1 above)
Comment (annevk) [reply dom/events/event-global.worker.js]: Yes it is, what if the callback throws?

Comment (foolip) [reply dom/events/event-global.worker.js]: The callback is invoked synchronously by `dispatchEvent` so the exception will reach the right place. Unless dispatchEvent suppresses exceptions and moves along to the next callback?

Comment (annevk) [reply dom/events/event-global.worker.js]: That's not how event listeners work. They use "report the exception" which crashes the harness.

Comment (foolip) [reply dom/events/event-global.worker.js]: Well there you go, I am just wrong :)

**Pull request merged**